### ISSUE TITLE
Create and test a NNM add-on

### ISSUE THREAD

**Issue by Idhrendur** [Idhrendur/paradoxGameConverters#38]
Converting into NNM would be amazing. There should be an add-on for people who would like to do so. My guess for the process is here: https://forum.paradoxplaza.com/forum/index.php?threads/the-eu4-to-v2-converter-project.707362/page-168#post-20401658

**Comment by derHochmeister** [comment]
It can't work unless you mix the countries.txt from both mods into one.

**Comment by derHochmeister** [comment]
New Nations Mod is not compatible with the converter, since both change countries and religions.

**Comment by kratostatic** [comment]
Well yes, the idea would be we would use NNM as the base mod rather than Vanilla and change what the converter changes as required.  Here's what Idhrendur says in the linked forum post:

> As well, the process should actually be something along the lines of:
1. Copy NNM into the blank_mod folder.
2. Update various mappings.
3. Convert game, which will create a combined mod.
4. Load the combined mod on its own.

Step two can probably be skipped and still produce a loadable mod, but it wouldn't take advantage of NNM's new nations. I've never tested the process before, though. But AFAIK it should work.

**Comment by derHochmeister** [comment]
I tried to do a compatibility for NNM. The result was Vic2 crashing because it says files were full of errors. Also, many NNM tags would be spammish and the history part of NNM would smash and corrupt with the converter's history files.

Sadly, NNM is not compatible.

**Comment by kratostatic** [comment]
What did you actually do exactly?  Did you just straight-up copy NNM into the blank mod/output folder?  Because except for pops the history folders are all empty.  If you delete the contents of the history folders in the "new" blank mod then it should work a lot better.

As for the tags, as I understand that's one of the main things that NNM actually changes so adding new tags would be a feature not a bug, although we can probably be a bit smarter in what we add in when we compare the original blank mod to NNM - some of the nations probably exist already.

I expect that some of the code will have to change anyway to make this work properly anyway - ultimately I'll need to have two different blank mods and that will mean the code needs to know where to look depending on the settings.

  I'll probably have a bit of play about with this myself tonight.

**Comment by derHochmeister** [comment]
No, no. I added all the new tags NNM has to the blankmod, same thing done for cultures and religions. I also checked for any possible overlap in the blankmod, didn't found any. And yet it'd crash.

**Comment by kratostatic** [comment]
Ok, thanks.

**Comment by kratostatic** [comment]
Right, I copied in the entire thing except for those history folders that were blank, and running the converter through it worked, producing a working (if very buggy) mod which had NNM decisions and events.  So good to know that this is actually possible, I was quite worried for a bit there!

**Comment by IohannesIohannium** [comment]
I've just run a test mod (default 1805 start save, enhanced by #517), editing the following:

-had to create a 2619 province history, as in Vanilla it is a Siberian wasteland province, but in NNM it is Ohrid
-had to disable history folder of NNM
-had to disable common/cultures.txt of NNM (it'd create a lot of "nocultures" for the Vanilla conversions)
-had to disable common/religions.txt of NNM (it'd create a lot of "noreligions" for the Vanilla conversions)

The yealded result is, in fact, playable, although probably not one of the best looking, full of gaps:

![image](https://user-images.githubusercontent.com/24550392/43907191-364377de-9bf5-11e8-992a-dacd0bae5135.png)


It seems, though, that NNM was renamed by its creators as "Blood and Iron mod".

The work to do still is how to merge the tags' files of NNM and of the conversions, whose differences can be summed up with this list of NNM tags (if there's a semi-colon, after that there is the tag the converter currently uses for the same country. Otherwise, it's not default in the conversion or never appears.): https://pastebin.com/HaJ8gZKH
